### PR TITLE
refactor(core): remove kratos registration endpoint

### PR DESCRIPTION
**Pull request by vindard**
## Description

Remove registration webhook and move account creation to login use-case

### CODE REVIEW

File: core/api/src/app/authentication/login.ts [hunk 1]
```diff
@@ -104,6 +106,12 @@ export const loginWithPhoneToken = async ({
     })
     if (kratosResult instanceof Error) return kratosResult
 
+    const account = await createAccountFromRegistrationPayload({
+      secret: KRATOS_CALLBACK_API_KEY,
```
Comment (vindard) [inline core/api/src/app/authentication/login.ts:110]: This is the only place we're applying the callback api key now, and it's pretty much redundant here. Should I remove?

File: dev/config/ory/kratos.yml [hunk 2]
```diff
@@ -132,19 +132,6 @@ selfservice:
             #         name: Authorization
             #         value: The-Value-of-My-Key
             #         in: header
```
Comment (nicolasburtey) [inline dev/config/ory/kratos.yml:134]: you could probably remove the 15 lines above also

Comment (vindard) [reply dev/config/ory/kratos.yml]: https://github.com/GaloyMoney/galoy/pull/3925/commits/ff8786f622994e59d6171cdcf2eac4bdf352432f

File: core/api/src/app/authentication/login.ts (hunk 1 above)
Comment (nicolasburtey) [reply core/api/src/app/authentication/login.ts:110]: yes `secret` can be removed completely, together with `KRATOS_CALLBACK_API_KEY` (also in charts/deployment)

Comment (nicolasburtey) [reply core/api/src/app/authentication/login.ts:110]: I think you could call `createAccountWithPhoneIdentifier` directly here

Comment (nicolasburtey) [reply core/api/src/app/authentication/login.ts:110]: ie: 

```
  const account = await createAccountWithPhoneIdentifier({
    newAccountInfo: { phone, kratosUserId: userId },
    config: getDefaultAccountsConfig(),
    phoneMetadata,
  })
```

Comment (vindard) [reply core/api/src/app/authentication/login.ts]: Resolved https://github.com/GaloyMoney/galoy/pull/3925/commits/ce81cf72bbd549e0434336b0642e9907a0164780

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**